src/Merge/ExtraPackage.php

```diff
@@ -165,6 +171,20 @@ public function mergeInto(RootPackageInterface $root, PluginState $state)
         }
 
         $this->mergeExtra($root, $state);
+
+        // Merge source reference information for merged packages.
+        // @see RootPackageLoader::load
+        $references = array();
+        foreach (array('require', 'require-dev') as $linkType) {
+            $linkInfo = BasePackage::$supportedLinkTypes[$linkType];
+            $method = 'get'.ucfirst($linkInfo['method']);
+            $links = array();
+            foreach ($root->$method() as $link) {
+                $links[$link->getTarget()] = $link->getConstraint()->getPrettyString();
+            }
+            $references = $this->extractReferences($links, $references);
+        }
+        $root->setReferences($references);
```

Comment:
I'd say that both are wanted. At the very least `unwrapIfNeeded` will cast or provide access to the aliased `RootPackage`. Upstreaming interface additions and tests is always a good idea I think. See https://github.com/composer/composer/commit/06c44ce998cdb0db2a522ec6086d6ce25bd79b4a for an example of us doing this previously.